### PR TITLE
Feat: customized color options

### LINKED ISSUE #234: [FR] Customized color options in color picker for Desktop
### Description

Expose the color options in `lib/src/editor/toolbar/utils/color_generators.dart ` to `FloatingToolbarStyle`.

### Impact

Developers are able to use their own colors for text and background(highlight) toolbar items. 

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by hyj1204**
To close #234 
Related to #211 

To customize color options: 

### Desktop
Give a list of `ColorOption` to `buildTextColorItem` and/or `buildHighlightColorItem` functions to display your personalized colors. 
```
       if (PlatformExtension.isDesktopOrWeb) {
            return FloatingToolbar(
              items: [
                buildTextColorItem(
                  colorOptions: [
                    ColorOption(colorHex: '164B60', name: 'Navy'),
                    ColorOption(colorHex: '1B6B93', name: 'Blue'),
                  ],
                ),
                buildHighlightColorItem(
                  colorOptions: [
                    ColorOption(colorHex: '4FC0D0', name: 'Light Blue'),
                    ColorOption(colorHex: 'A2FF86', name: 'Green'),
                  ],
                ),
              ],
          ...
```
image
image

### Mobile
Give a list of text/background `ColorOption` to `buildTextAndBackgroundColorMobileToolbarItem` to display your personalized colors. 

```
MobileToolbar(
                  editorState: editorState,
                  toolbarItems: [
                    buildTextAndBackgroundColorMobileToolbarItem(
                      textColorOptions: [
                        ColorOption(colorHex: '164B60', name: 'Navy'),
                        ColorOption(colorHex: '1B6B93', name: 'Blue'),
                      ],
                      backgroundColorOptions: [
                        ColorOption(colorHex: '4FC0D0', name: 'Light Blue'),
                        ColorOption(colorHex: 'A2FF86', name: 'Green'),
                      ],
                    ),
                  ],
                ),
```
image

image

### CODE REVIEW

**Comment by LucasXu0** [comment]
@hyj1204 Please fix the CI issues.

File: lib/src/editor/toolbar/desktop/items/color/highlight_color_toolbar_item.dart [hunk 1]
```diff
@@ -1,34 +1,37 @@
...
-  isActive: onlyShowInTextType,
-  builder: (context, editorState) {
-    String? highlightColorHex;
+ToolbarItem buildHighlightColorItem({List<ColorOption>? colorOptions}) {
+  return ToolbarItem(
+    id: 'editor.highlightColor',
+    group: 4,
+    isActive: onlyShowInTextType,
+    builder: (context, editorState) {
+      String? highlightColorHex;
```
Comment (Xazin) [inline lib/src/editor/toolbar/desktop/items/color/highlight_color_toolbar_item.dart:10]: This seems to be unused

Comment (Xazin) [inline lib/src/editor/toolbar/desktop/items/color/highlight_color_toolbar_item.dart:10]: Does this ever get assigned any other value than null? Seems redundant

Comment (hyj1204) [reply lib/src/editor/toolbar/desktop/items/color/highlight_color_toolbar_item.dart:10]: `highlightColorHex` is used to show a checked mark on current color options(if it has a color), I assigned value to it now(same with text color).
image

Nice catch, Thanks!

**Pull request merged**